src/en/gogoanime/src/eu/kanade/tachiyomi/animeextension/en/gogoanime/GogoAnime.kt

```diff
@@ -118,35 +116,15 @@ class GogoAnime : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
     override fun videoUrlParse(document: Document) = throw Exception("not used")
 
     override fun List<Video>.sort(): List<Video> {
-        val quality = preferences.getString("preferred_quality", "1080")
-        if (quality != null) {
-            val newList = mutableListOf<Video>()
-            var preferred = 0
-            for (video in this) {
-                when {
-                    video.quality.contains(quality) -> {
-                        newList.add(preferred, video)
-                        preferred++
-                    }
-                    abs(qualityToInt(video.quality) - quality.toInt()) < 100 -> {
-                        newList.add(preferred, video)
-                    }
-                    else -> {
-                        newList.add(video)
-                    }
-                }
-            }
-            return newList
-        }
-        return this
-    }
-
-    private fun qualityToInt(quality: String): Int {
-        return try {
-            quality.filter { it.isDigit() }.toInt()
-        } catch (e: NumberFormatException) {
-            -100
-        }
+        val quality = preferences.getString("preferred_quality", "1080")!!
+        val server = preferences.getString("preferred_server", "Gogostream")!!
+
+        return this.sortedWith(
+            compareBy(
+                { it.quality.contains(quality) },
+                { it.quality.contains(server) }
+            )
+        ).reversed()
```

Comment:
POV: You are someone who actually knows Kotlin.
😂 thank you for fixing my terrible code